### ISSUE TITLE
Customize key shortcut band

### ISSUE THREAD

**Issue by ulvund** [blinksh/blink#227]
Can I add keys to the shortcut band in a new version of blink? Like : for vim navigation?

**Comment by goerz** [comment]
What is the shortcut band?

**Comment by carloscabanero** [comment]
We have this in our backlog. The idea is that the central part of smart keys can be configured for other applications. So you can send emacs or vim commands when you are on them, just by tapping, instead of doing the keyboard combo.

This is still in the backlog and we have a few things to work out before, but I will make sure Blink does that ;)

**Comment by goerz** [comment]
Oh, now I get it! Very cool!

**Comment by drbarrett** [comment]
One more vote for this, or even just add the : to the smart key list

**Comment by Parakleta** [comment]
This is more relevant now that Apple has added the pull for punctuation feature on their keyboard.  A lot of the characters in the smart key list are trivial to access directly (in fact all of them are available as 'pull-down' characters on the alphabetic or number/symbol keyboards), but I cannot find a way to access the backtick (grave) character efficiently.

**Comment by carloscabanero** [comment]
We have a branch that includes this, but unfortunately I'm not able to prioritise this yet. So lets add them directly to the SmartKeys bar. Other than the : and the backtick, any other characters you would like me to add?

**Comment by Parakleta** [comment]
I don’t know if there’s a resource that shows all of the iPad/iPhone keyboards in each language, but the priority should be those keys not on the alphabetic or numeric/punctuation shifted keyboard (i.e. directly available, or one shift away).  On the iPad English keyboard, `[,.!?]` is on alphabetic, and `[-@#$&*()'"%+=/;:]` on the numeric/punctuation keyboard, so this leaves ```[<>[\]{}\\^_`|~]``` as not trivially accessible (12 characters in total).

Obviously TAB and ESC as control characters are quite useful, but I’m not sure why currently one is within the shortcut band and one without.

**Comment by mladenmarkov** [comment]
Can’t we have these configurable? I frequently use the backslash and dot, along with tab in blink on an iPhone, and I’d like to have those closely together.

**Comment by tbrowder** [comment]
I regularly need the back-tick char (left single quotation mark) and can't find it anywhere.

**Comment by goerz** [comment]
If you're using the standard iOS software keyboard, tap and hold the `'` key (on the "123" keyboard pane). This gives you a choice between different single quotes/accents, as well as the backtick.

**Comment by yury** [comment]
Also, you can tap and hold `123` button, then move to `'` and force touch it :)

https://www.youtube.com/watch?v=IoO1k7kVIWA

**Comment by tbrowder** [comment]
On Thu, Apr 5, 2018 at 13:37 Yury Korolev <notifications@github.com> wrote:

> Also, you can tap and hold 123 button, then move to ' and force touch it
> :)
>
> https://www.youtube.com/watch?v=IoO1k7kVIWA
>

Works great! Thanks.

-Tom

**Comment by maurizio-manuguerra-mq** [comment]
> Can’t we have these configurable? I frequently use the backslash and dot, along with tab in blink on an iPhone, and I’d like to have those closely together.

Is this going to happen? The possibility to customize the smartkeys bar seems like an obvious feature. As it is, it is not very helpful for a variety of use cases.

**Comment by sfatula** [comment]
This would be extremely useful and productive

**Comment by jonwillpoe** [comment]
Any outlook on this? Would be very helpful for me

**Comment by yury** [comment]
Hi @jonwillpoe.

Just for me, what you want to add/remove there?

**Comment by jonwillpoe** [comment]
Hello @yury I’d like to have a “:” (colon), replace backslash with forward slash, and replace brackets with parentheses.

**Comment by yury** [comment]
You don’t have them in swipe access on main kb?

Can you screenshot your kb?

**Comment by jonwillpoe** [comment]
![IMG_0112](https://user-images.githubusercontent.com/57763671/81855255-1b299b80-952d-11ea-82b0-f216fc4f7399.jpeg)

**Comment by yury** [comment]
Yep, you have them all on main kb. I don’t know, but I’d train muscle memory on default kb

**Comment by jonwillpoe** [comment]
Good enough (actually didn’t even know you could swipe down to access layers). Thank you for your replies. If possible, I still think it would also be useful to change the position of the shortcut band keys and even the keys themselves at some point.
